data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessorService.java

```diff
@@ -40,32 +41,35 @@ public class GeoIPProcessorService {
     private static final Logger LOG = LoggerFactory.getLogger(GeoIPProcessorService.class);
     public static final String DATABASE_1 = "first_database_path";
     public static final String DATABASE_2 = "second_database_path";
+    private static final String TEMP_PATH_FOLDER = "GeoIP";
     private GeoIPProcessorConfig geoIPProcessorConfig;
     private LicenseTypeOptions licenseType;
     private GetGeoData geoData;
-    private List<DatabasePathURLConfig> databasePath;
+    private List<String> databasePaths;
     private final String tempPath;
     private final ScheduledExecutorService scheduledExecutorService;
     private final DBSourceOptions dbSourceOptions;
+    private final MaxMindConfig maxMindConfig;
     public static volatile boolean downloadReady;
     private boolean toggle;
     private String flipDatabase;
 
 
     /**
      * GeoIPProcessorService constructor for initialization of required attributes
-     * @param geoIPProcessorConfig geoIPProcessorConfig
-     * @param tempPath tempPath
+     *
+     * @param geoIpServiceConfig geoIpServiceConfig
      */
-    public GeoIPProcessorService(GeoIPProcessorConfig geoIPProcessorConfig, String tempPath) {
+    public GeoIPProcessorService(final GeoIpServiceConfig geoIpServiceConfig) {
```

Comment:
I think this is going to be provided by the extension in the end, right?

It would be good to move this into the `.extension` package. That could happen in a follow-on PR.

Reply:
Yes, it's provided by extension supplier. I will move it to right package.

Reply:
QUES: do we know if this will potentially be used in other pipeline plugins? Need to catch up on justification why we put it in extensions

Reply:
By "number of processors" are we expecting future processor plugin to depend upon this? It would be good to have examples but not blocking this PR.

Reply:
I mean any number of `geoip` processors.

```
pipeline-a:
  processor:
    - geoip:
    - geoip:
pipeline-b:
  processor:
    - geoip:
```

All of these will share the same underlying objects and resources from Maxmind.